workflow/cron/operator.go

```diff
@@ -178,6 +184,89 @@ func (woc *cronWfOperationCtx) patch(ctx context.Context, patch map[string]inter
 	}
 }
 
+// TODO: refactor shouldExecute in steps.go
+func shouldExecute(when string) (bool, error) {
+	if when == "" {
+		return true, nil
+	}
+	expression, err := govaluate.NewEvaluableExpression(when)
+	if err != nil {
+		return false, err
+	}
+
+	result, err := expression.Evaluate(nil)
+	if err != nil {
+		return false, err
+	}
+
+	boolRes, ok := result.(bool)
+	if !ok {
+		return false, argoerrs.Errorf(argoerrs.CodeBadRequest, "Expected boolean evaluation for '%s'. Got %v", when, result)
+	}
+	return boolRes, nil
+}
+
+func evalWhen(cron *v1alpha1.CronWorkflow) (bool, error) {
+	if cron.Spec.When == "" {
+		return true, nil
+	}
+	cronBytes, err := json.Marshal(cron)
+	if err != nil {
+		return false, err
+	}
+
+	m := make(map[string]interface{})
+	addSetField := func(name string, value interface{}) {
+		m[fmt.Sprintf("cronworkflow.%s", name)] = value
+	}
+
+	addSetField("name", cron.Name)
+	addSetField("namespace", cron.Namespace)
+	addSetField("labels", cron.Labels)
+	addSetField("annotations", cron.Labels)
+
+	labelsStr, err := json.Marshal(&cron.Labels)
+	if err != nil {
+		// We shouldn't hit this
```

Comment:
I'd replace this with a lint ignore, similar to https://github.com/argoproj/argo-workflows/pull/12917, as that simplifies the code

Reply:
Sure I can do this.

Reply:
Actually we shouldn't do this, everywhere we Marshal we still error check.

Reply:
Then the comment should be removed -- it's really one or the other

Reply:
Will remove the comment in that case, we should be consistent in logic even if we expect marshalling to always  succeed.

Reply:
If it has a comment, I don't think we need consistency, given that's what the comment explains. The lint ignores I wrote in #12917 are followed by comments (as is best practice for any lint ignore)